gradle/libs.versions.toml

```diff
@@ -141,6 +142,8 @@ androidx-constraintlayout-compose = { group = "androidx.constraintlayout", name
 androidx-constraintlayout-main = { group = "androidx.constraintlayout", name = "constraintlayout", version.ref = "androidx-constraintlayout-main" }
 androidx-core-ktx = { group = "androidx.core", name = "core-ktx", version.ref = "androidx-core" }
 androidx-core-main = { group = "androidx.core", name = "core", version.ref = "androidx-core" }
+androidx-credentials = { module = "androidx.credentials:credentials", version.ref = "androidx-credentials" }
```

Comment:
Sure, I'll follow this convention, but I'm unsure if I see a value in adding this suffix - what's the problem of having simply `libs.androidx.core` instead of `libs.androidx.core.main`?

Reply:
Great question @wzieba ! 💯 

I tried to summarize that in those version catalogs migration PRs of mine ([example](https://github.com/wordpress-mobile/WordPress-Android/pull/21262)):

> Where needed the word main is added in its suffix to make sure that a dependency cannot be a parent of another dependency and a dependency itself. This helps in various corner cases.

I didn't want to complicate things and try to give such corner cases, but there are a few. For example, consider this line [here](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/build.gradle#L525) (`libs.androidx.test.espresso.core.get().version`). If you go to the version catalog itself and remove the `-core` suffix, then try to sync, you will immediately notice the following build error:

```
No signature of method: org.gradle.accessors.dm.LibrariesForLibs$AndroidxTestEspressoLibraryAccessors.get() is applicable for argument types: () values: []
```

So, although defining `androidx-test-espresso` to point to `espresso-core` and `androidx-test-espresso-contrib` to point to `espresso-contrib` is perfectly fine when those are using as dependencies, when you start trying to do something more, then you get into trouble.

PS: Think about this `-core` being pretty similar to `-main`, it is just that sometimes you don't have this `core` naming convention on a dependency signature. Initially, I started naming everything with the `-core` suffix, but then I realized that some dependencies have both a main dependency and a core dependency (see [androidx-lifecycle-livedata](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/gradle/libs.versions.toml#L148C1-L149)), plus no dependency have the `main` naming convention, so it was free for me to pick-up  and thus I went with this naming convention (instead of `core`). Also see this [androidx-core](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/gradle/libs.versions.toml#L142C1-L143) and imagine having to name this as `androidx-core-core` (😅).

Does this help? 🙏

Reply:
> If you go to the version catalog itself and remove the -core suffix

Sorry, I'm not following - why would I want to remove `-core` suffix, if it's in the name of dependency?

What I mean is that I don't see value in adding artificial suffixes, meaning suffixes that are not part of a dependency name.

`androidx-test-espresso-core` is fine to me because module is `androidx.test.espresso:espresso-core`. 
`androidx-credentials-main` is not good idea IMO, as module is `androidx.credentials:credentials` so "main" came out of nowhere.

Reply:
Ah ok, I get it: `libs.androidx.test.espresso` points to the group accessor, instead of the specific dependency.

Reply:
Yea, apologies for trying to explain it in such a complicated way, I could have definitely done a better job, I was just trying to find a example that makes sense and you can quickly see the build erroring.... 🫣